src/components/Button/Button.js

```diff
@@ -1,36 +1,42 @@
-import styled from 'styled-components';
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { func, node, string } from 'prop-types';
+import { useSpring } from 'react-spring';
 
-import {
-  colorWhite,
-  colorGreen,
-  colorDarkGreen,
-  fontLarge,
-  fontWeightBold,
-  borderRadius,
-  spacingMedium
-} from '../../styles/designTokens';
+import StyledButton from './StyledButton';
 
-const Button = styled.button`
-  position: relative;
-  display: flex;
-  align-items: center;
-  text-decoration: none;
-  background-color: ${colorGreen};
-  color: ${colorWhite};
-  padding: 0 ${spacingMedium};
-  margin: ${spacingMedium} 0;
-  border: 0;
-  border-bottom: 0.3rem solid ${colorDarkGreen};
-  border-radius: ${borderRadius};
-  font-size: ${fontLarge};
-  font-weight: ${fontWeightBold};
-  text-align: left;
-  width: 100%;
-  height: 5rem;
+function Button({ to, onClick = () => {}, children, ...props }) {
+  const mappedProps = {
+    ...props,
+    onClick,
+    ...(to ? { as: Link, to } : { onClick })
+  };
 
-  &:not(:disabled) {
-    cursor: pointer;
-  }
-`;
+  const [isClicked, setClicked] = useState(false);
+  const animationStyles = useSpring({
+    borderBottomWidth: isClicked ? 0 : 3,
+    config: {
+      tension: 500,
+      friction: 8
+    }
+  });
+
+  return (
+    <StyledButton
+      {...mappedProps}
+      style={animationStyles}
+      onMouseDown={() => setClicked(true)}
```

Comment:
Maybe use similar approach as in one of previous animations with **onRest** callback?

Reply:
Using this approach holding click on button won't work. Trigger of reverse animation is strictly connected with MouseUp event which affects positively the animation timing.